### ISSUE TITLE
Idle session disconnection doesn't work

### ISSUE THREAD

**Issue by metalefty** [neutrinolabs/xorgxrdp#141]
Original issue: https://github.com/neutrinolabs/xrdp/issues/1297

On some environment, idle time is not detected properly. 
```
[500052.388] rdpDeferredIdleDisconnectCallback: time has gone backwards, resetting
```

**Comment by pieska** [comment]
#142 

please review and test, fixed the issue for me

**Comment by pieska** [comment]
#143 

or this optimized version with a minimum of timer executions

**Comment by metalefty** [comment]
Fixed by #143
